### ISSUE TITLE
Add Redmine as a data source for tasks

### ISSUE THREAD

**Issue by sputtene** [johannesjo/super-productivity#755]
## Problem Statement
I use Redmine as an issue tracker/project management system for some of my projects.

It would be great to have Redmine integration next to Gitlab, Github, &hellip; integration as a source for issues and tasks.

## :grey_question: Possible Solution
Add support for Redmine as a task data source.

Redmine has a REST API, so using it to fetch issues and their descriptions should not be very difficult.

## :arrow_heading_up: Describe alternatives you've considered
Short of migrating away from Redmine (which is not a viable option), I'm stuck with copy-pasting task descriptions and have no real integration between Super Productivity and the actual issue tracker.

## :heavy_plus_sign: Additional context
I wouldn't mind coding up the integration myself (I'm not a Typescript programmer, but I'm willing to learn), but I have no real idea where to start.

I also haven't found any "so you want to add integration with another backend" technical documentation, so knowing where to start and what information to fetch from Redmine for a full integration with Super Productivity would mean I'd have to reverse engineer for example the Github integration and mimic its behaviour using the Redmine API instead of the Github one.

In short, I would like to just send a pull request for this feature, but I'll need some hand holding.

**Comment by johannesjo** [comment]
Hey there! Thanks for opening this up! Best place to start is probably by looking at the other issue provider integrations under `src/app/features/issue/`. It's probably a lot to take in if you're not familiar with neither TypeScript nor Angular... But please let me know if you have any questions!

**Comment by sputtene** [comment]
Hi,


I'm trying to get the project running on my laptop according to the build instructions in the Contributing section of the README, but I'm already running into a problem.

I've cloned the repo, but when trying to run `npm install` to install the dependencies, I get this error:
```
offbyone@snowwhite super-productivity % npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: superProductivity@6.3.2
npm ERR! Found: @angular/core@10.2.4
npm ERR! node_modules/@angular/core
npm ERR!   peer @angular/core@">=8.1.3 <=10.999.9" from angular-material-css-vars@1.1.2
npm ERR!   node_modules/angular-material-css-vars
npm ERR!   peer @angular/core@"10.2.4" from @angular/common@10.2.4
npm ERR!   node_modules/@angular/common
npm ERR!     peer @angular/common@">=8.1.3 <=10.999.9" from angular-material-css-vars@1.1.2
npm ERR!     node_modules/angular-material-css-vars
npm ERR!     peer @angular/common@"^10.0.0 || ^11.0.0-0" from @angular/material@10.2.7
npm ERR!     node_modules/@angular/material
npm ERR!       peer @angular/material@">=8.1.3 <=10.999.9" from angular-material-css-vars@1.1.2
npm ERR!       node_modules/angular-material-css-vars
npm ERR!   1 more (@angular/material)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/core@"11.0.5" from @angular/animations@11.0.5
npm ERR! node_modules/@angular/animations
npm ERR!   peer @angular/animations@"^10.0.0 || ^11.0.0-0" from @angular/material@10.2.7
npm ERR!   node_modules/@angular/material
npm ERR!     peer @angular/material@">=8.1.3 <=10.999.9" from angular-material-css-vars@1.1.2
npm ERR!     node_modules/angular-material-css-vars
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

I have no experience using `npm`, I also have no idea on how to resolve this dependency conflict.

There is a full report saved and a debug log as well. I can upload these files if that helps with solving the issue. 

I have programming experience, just not in TypeScript or Angular. I do know some JavaScript, HTML, CSS, ... so I guess once I get my development environment set up, I'll be able to get something working without too much help :)

**Comment by johannesjo** [comment]
Hey hey! You might want to try yarn instead of npm. The project uses it as default package manager (indicated by the yarn.lock file in the root dir). I would assume that this would fix the problem.

It's great that you are up for the challenge. Just to give you another warning and what to look out for: Another important technology being used in the project are observables (via rxjs). This is some tough cookie chew one (for most) if you're not familiar with it, but I personally think that there is a lot of beauty in it once you get the hang of it :)

**Comment by sputtene** [comment]
Hi,


I've installed `yarn`. Running `yarn install` in the super-productivity top-level directory, yields another error:
```
offbyone@snowwhite super-productivity % yarn install
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
warning Resolution field "webpack@4.44.2" is incompatible with requested version "webpack@4.44.1"
[2/4] 🚚  Fetching packages...
error jasmine@3.6.3: The engine "node" is incompatible with this module. Expected version "^10 || ^12 || ^14". Got "15.5.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I'll look into rxjs and observables. It's not something I'm familiar with, but after reading https://rxjs.dev/guide/observable it seems like an elegant way to structure data flows.

**Comment by mkpaz** [comment]
@sputtene Both of this problems probably because of NodeJS version you're using. Not sure about jasmine, but [here](https://github.com/angular/angular/blob/d7665fdc96e30a21cf33c96fb0cc29a90404b5f9/package.json#L11) is the range of supported NodeJS version for angular. Try to use lower NodeJS version. I had zero problems following build instructions recently.

```
$ node -v
v12.18.4
```

**Comment by sputtene** [comment]
Hi,


Long story short: it works!

I have used `nvm` to install the same node version as you (12.18.4), which installed. I got some security and deprecation warnings though (I can paste those if you want to look into them?).
Installing the @angular/cli thing worked as well and I can now access http://localhost:4200/ after running `ng start`.

I still have to familiarize myself with the environment, the development tools (you mention redux dev tools) and the language, but at least I can start playing around with the code now.

I'll let you know when I have made some progress.

**Comment by johannesjo** [comment]
@sputtene that's great news! Let me know if something comes up!

@mkpaz thank you very much for helping out!
